### ISSUE TITLE
[BUG] v4.0.3 foreign key on update

### ISSUE THREAD

**Issue by tsvetiligo** [bizley/yii2-migration#100]
Yii v2.0.35
Yii 2 Migration Generator Tool v4.0.3
MySQL 5.7

Migration tool generates dropForeignKey() and addForeignKey() for same key, then try drop index which don't exists.

```
<?php

use yii\db\Migration;

class m200519_132335_test1 extends Migration
{

    public function up()
    {
        $tableOptions = null;
        if ($this->db->driverName === 'mysql') {
            $tableOptions = 'CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci ENGINE=InnoDB';
        }

        $this->createTable('{{%test1}}', [
            'col1' => $this->bigPrimaryKey()->unsigned(),
            ], $tableOptions);

        $this->createTable('{{%test2}}', [
            'col1' => $this->bigPrimaryKey()->unsigned(),
            'col2' => $this->bigInteger()->unsigned()->notNull(),
            ], $tableOptions);

        $this->createIndex('col2_index', '{{%test2}}', 'col2');
        $this->addForeignKey('col2_foreign', '{{%test2}}', 'col2', '{{%test1}}', 'col1', 'RESTRICT', 'RESTRICT');
    }

    public function down()
    {
        $this->dropTable('{{%test1}}');
        $this->dropTable('{{%test2}}');
    }
}
```
```
./yii migrate
Yii Migration Tool (based on Yii v2.0.35)

Total 1 new migration to be applied:
	m200519_132335_test1

Apply the above migration? (yes|no) [no]:yes
*** applying m200519_132335_test1
    > create table {{%test1}} ... done (time: 0.015s)
    > create table {{%test2}} ... done (time: 0.014s)
    > create index col2_index on {{%test2}} (col2) ... done (time: 0.012s)
    > add foreign key col2_foreign: {{%test2}} (col2) references {{%test1}} (col1) ... done (time: 0.032s)
*** applied m200519_132335_test1 (time: 0.095s)


1 migration was applied.

Migrated up successfully.
```
```
./yii migration/update "*"
Yii 2 Migration Generator Tool v4.0.2
 > 1 table excluded by the config.
 > Are you sure you want to generate migrations for the following tables?
   - test1
   - test2 (yes|no) [no]:yes

 > Comparing current table 'test1' with its migrations ...TABLE IS UP-TO-DATE.

 > Comparing current table 'test2' with its migrations ...DONE!

 > Generating migration for updating table 'test2' ...DONE!
 > Saved as '/srv/migrations/m200618_170953_update_table_test2.php'

 Generated 1 file
 (!) Remember to verify files before applying migration.
```
```
<?php

use yii\db\Migration;

class m200618_170953_update_table_test2 extends Migration
{
    public function up()
    {
        $this->dropForeignKey('col2_foreign', '{{%test2}}');

        $this->addForeignKey(
            'col2_foreign',
            '{{%test2}}',
            ['col2'],
            '{{%test1}}',
            ['col1'],
            'RESTRICT',
            'RESTRICT'
        );

        $this->dropIndex('col2_foreign', '{{%test2}}');
    }

    public function down()
    {
        $this->createIndex('col2_foreign', '{{%test2}}', ['col2']);

        $this->dropForeignKey('col2_foreign', '{{%test2}}');

        $this->addForeignKey(
            'col2_foreign',
            '{{%test2}}',
            ['col2'],
            '{{%{{%test1}}}}',
            ['col1'],
            'RESTRICT',
            'RESTRICT'
        );
    }
}
```

**Comment by tsvetiligo** [comment]
Console says "Yii 2 Migration Generator Tool v4.0.2" but I am sure it v4.0.3.

**Comment by bizley** [comment]
You're on fire, man :)
Although I'm sorry all these bugs went through undetected before.
This is interesting case because there probably may be some conflict in detecting proper index when two have been set for the same column (one directly, second from fk) but I need to investigate.
And yeah, forgot to bump version in 4.0.3.

**Comment by tsvetiligo** [comment]
In MySQL, addForeignKey () for a column without an index creates a new index with the same name as the foreign key, and the index is not deleted when the foreign key is deleted, so I prefer to add the index manually before creating the foreign key to separate them.

**Comment by tsvetiligo** [comment]
Another minor thing is if the update generates more than one file, the first file does not have a prefix (“01”), but all subsequent ones have (“02” and so on).

**Comment by bizley** [comment]
Please report it separately.

**Comment by bizley** [comment]
Fixed in [4.0.4](https://github.com/bizley/yii2-migration/releases/tag/4.0.4)
